app/src/main/java/de/westnordost/streetcomplete/quests/LastPickedValuesStore.kt

```diff
@@ -34,13 +34,13 @@ private const val MAX_ENTRIES = 100
  *  items of the sequence, in their original order.
  * If there are fewer than `target` unique items, continues counting items
  *  until that many are found, or the end of the sequence is reached.
- * If the first item (i.e. the most recent one) is not null, it is always
- *  included, displacing the least-common of the other items if necessary.
+ * If the `first` most recent items are not null, they are always included,
+ *  displacing the least-common of the other items if necessary.
  */
-fun <T : Any> Sequence<T?>.mostCommonWithin(target: Int, historyCount: Int): Sequence<T> {
+fun <T : Any> Sequence<T?>.mostCommonWithin(target: Int, historyCount: Int, first: Int): Sequence<T> {
```

Comment:
I wonder if this signature would be more understandable as
```kt
fun <T : Any> Sequence<T?>.mostCommonWithin(first: Int, top: Int, within: Int): Sequence<T> {
```
So instead of `mostCommonWithin(6, 30, 1)` you'd do `mostCommonWithin(1, 5, 30)`.
I guess it's unintuitive in the edge case of "what if the first entry is null and we return 6 top values instead".